inst/include/sampler.h

```diff
@@ -403,7 +413,71 @@ namespace smc {
     {
         return expl(2*stableLogSumWeights(pPopulation.GetLogWeight())-stableLogSumWeights(2.0*pPopulation.GetLogWeight()));
     }
+    template <class Space, class Params>
+    std::vector<population<Space>> sampler<Space,Params>::GetApop(void) const
+    {
+        long N = GetNumber();
+        long lHistorylength = GetHistoryLength();
+        int historylength = static_cast<int>(lHistorylength);
+        /// Output container for particle population ancestor lines
+        std::vector<population<Space> > populationAL(historylength);
+        /// Get final iteration indices
+        arma::Col<unsigned int> AI_tmp = History[historylength-1].GetAindices();
+        /// Get final iteration particle values
+        populationAL[historylength-1] = GetHistoryPopulation(historylength-1);
+        /// Iterating from last iteration to initiliaization period whereby:
+        /// 1. Re-ordering particle population elements for each t=0,...,T-1 accoridng to ancestral lines
+        /// 2. Storing these particle population classes in output: populationAL
+        if (historylength > 1){
+            for (int t = historylength - 2; t >= 0; --t){
+                for(int i = 0; i < static_cast<int>(N); ++i){

```

Comment:
This looks like a bad idea.

If we use a long for `N` to allow for large particle sets on architectures with small words or similar then we should really be using a long for indices which iterate over the particle set shouldn't we?

If there's a reason to strong prefer int then there's probably no harm in changing the type of N from long to int given the architectures and compilers in current used. There aren't many 16 bit `int`s around any more.

Reply:
I also find `uint64_t and `int64_t` more obvious, I think C++ pros also like size_t for the indices. 

```r
> Rcpp::evalCpp("sizeof(long)")    
[1] 8 
> Rcpp::evalCpp("sizeof(int)") 
[1] 4   
> Rcpp::evalCpp("sizeof(int64_t)")      
[1] 8  
> Rcpp::evalCpp("sizeof(size_t)")    
[1] 8
> 
```

Reply:
Changed!